streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java

```diff
@@ -138,6 +138,7 @@ public void registerThread(final String threadName) {
 
     public void unregisterThread(final String threadName) {
         threadVersions.remove(threadName);
+        maybeNotifyTopologyVersionWaitersAndUpdateThreadsTopologyVersion(threadName);
```

Comment:
Do we really need this here?

Reply:
Yeah, sorry I should have left an explanation for this -- basically if we have `#add/removeNamedTopology` futures blocked on this one thread ack'ing the topology update, and that update occurs between when the thread checks for these updates and when it checks for/goes into shutdown, then we need to update the waiters and potentially complete any that have just been waiting on this thread which has missed the update before shutting down